File: data structures and libraries/1062.py
while True:
    nVagoes = int(input())
    if nVagoes == 0:
        break
    
    while True:
        saida = input().strip()
        if saida == "0":
            break
        
        saida = list(map(int, saida.split()))
        entrada = list(range(1, nVagoes + 1))
        estacao = []
        j = 0
        possivel = True
        
        for vagon in entrada:
            estacao.append(vagon)
            
            while estacao and j < len(saida) and estacao[-1] == saida[j]:
                estacao.pop()
                j += 1
        
        # Verifica se todos os elementos de saida foram encontrados
        if j == len(saida):
            print("Yes")
        else:
            print("No")
    print("")

File: data structures and libraries/1062.cpp
#include <iostream>
#include <stack>
#include <vector>
#include <sstream>
#include <string>

using namespace std;

int main() {
    while (true) {
        int nVagoes;
        cin >> nVagoes;
        cin.ignore();

        if (nVagoes == 0) {
            break;
        }

        while (true) {
            string saida;
            getline(cin, saida);
            if (saida == "0") {
                break;
            }

            stringstream ss(saida);
            vector<int> saidaList;
            int num;
            while (ss >> num) {
                saidaList.push_back(num);
            }

            vector<int> entrada(nVagoes);
            for (int i = 0; i < nVagoes; ++i) {
                entrada[i] = i + 1;
            }

            stack<int> estacao;
            int j = 0;
            bool possivel = true;

            for (int vagon : entrada) {
                estacao.push(vagon);

                while (!estacao.empty() && j < saidaList.size() && estacao.top() == saidaList[j]) {
                    estacao.pop();
                    j++;
                }
            }

            if (j == saidaList.size()) {
                cout << "Yes" << endl;
            } else {
                cout << "No" << endl;
            }
        }

        cout << endl;
    }

    return 0;
}
